### PR TITLE
restructure/cleanup of iontests.js to enable good/bad/event skiplists;  all non-skipped test files pass

### PR DESCRIPTION
**Pull request by pbcornell**
Part of an effort to achieve a successful build of the development branch and appease reviewers.  ;)  Plus general cleanup of this file.

Lots of changes;  it may be helpful to use the "View file" option off the ellipsis (...) menu below.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/unit/iontests.js [hunk 1]
```diff
@@ -142,88 +82,280 @@ define(['intern', 'intern!object', 'intern/dojo/node!fs', 'intern/dojo/node!path
...
+        skipList[path] = 1;
+    });
+
+    // additional, known bad/slow test files:
+    skipList['ion-tests/iontestdata/good/subfieldVarUInt32bit.ion'] = 1;
+
+    return skipList;
+}
+
+// skiplists generated by toSkipList.pl
```
Comment (therapon) [inline tests/unit/iontests.js]: Is this comment referring to the function `toSkipList`? The `.pl` is what is confusing me here.

File: tests/unit/iontests.js [hunk 2]
```diff
@@ -142,88 +82,280 @@ define(['intern', 'intern!object', 'intern/dojo/node!fs', 'intern/dojo/node!path
...
+    'ion-tests/iontestdata/good/timestamp/equivTimeline/timestamps.ion',
+    'ion-tests/iontestdata/good/timestamp/leapDay.ion',
+    'ion-tests/iontestdata/good/timestamp/timestamp2011-02-20T19_30_59_100-08_00.10n',
+    'ion-tests/iontestdata/good/timestamp/timestamp2011-02.10n',
+    'ion-tests/iontestdata/good/timestamp/timestamp2011.10n',
+    'ion-tests/iontestdata/good/timestamp/timestampWithTerminatingEof.ion',
+    'ion-tests/iontestdata/good/timestamp/timestamps.ion',
+]);
+
+// /skiplists generated by toSkipList.pl
```
Comment (zslayton) [inline tests/unit/iontests.js]: Nit: stray `/`

File: tests/unit/iontests.js (hunk 1 above)
Comment (pbcornell) [reply tests/unit/iontests.js]: it's a perl script i have locally;  removed reference to avoid confusion

File: tests/unit/iontests.js (hunk 2 above)
Comment (pbcornell) [reply tests/unit/iontests.js]: that `/` was intended to indicate an end marker (similar to `/` usage in html markup);  removed/clarified

File: tests/unit/iontests.js [hunk 3]
```diff
@@ -142,88 +82,282 @@ define(['intern', 'intern!object', 'intern/dojo/node!fs', 'intern/dojo/node!path
...
+    'ion-tests/iontestdata/good/intBigSize13.10n',
+    'ion-tests/iontestdata/good/intBigSize14.10n',
+    'ion-tests/iontestdata/good/intBigSize16.10n',
+    'ion-tests/iontestdata/good/intBigSize256.10n',
+    'ion-tests/iontestdata/good/intBigSize256.ion',
+    'ion-tests/iontestdata/good/intBigSize512.ion',
+    'ion-tests/iontestdata/good/intBinary.ion',
+    'ion-tests/iontestdata/good/intLongMaxValuePlusOne.10n',
+    'ion-tests/iontestdata/good/intLongMinValue.10n',
+    'ion-tests/iontestdata/good/integer_values.ion',
```
Comment (wesboyt) [inline tests/unit/iontests.js]: comments from the previous skiplist should be moved over instead of left as a copy paste below

Comment (pbcornell) [reply tests/unit/iontests.js]: i don't have an easy way to do this.  current plan is to regenerate the skiplists with a perl script on a weekly basis

**Pull request merged**